### ISSUE TITLE
Unable to DL using E-H 

### ISSUE THREAD

**Issue by Cupbox** [happypandax/plugins#34]
While Chaika and NH work fine E-H does not work for me

**To Reproduce**
Login using EHentai Login
Clicking download gallery using the Firefox extension gives the following in the log
[---INFO---]--2019-Oct-30 21:57:22-: Finished handling item 
[---INFO---]--2019-Oct-30 21:57:22-: Failed to import item: BadZipFile("File is not a zip file",) 
[---INFO---]--2019-Oct-30 21:57:21-: Waiting on post-processing from "exhentai"... 
[---INFO---]--2019-Oct-30 21:57:15-: Downloading "2" requests for https://exhentai.org/g/1509162/87cf84cdae/ 
[---INFO---]--2019-Oct-30 21:57:15-: Received 2 requests from "exhentai" 
[---INFO---]--2019-Oct-30 21:57:10-: Querying resources from "exhentai"... 
[---INFO---]--2019-Oct-30 21:57:09-: Handling url https://exhentai.org/g/1509162/87cf84cdae/ 
[---INFO---]--2019-Oct-30 21:57:09-: Adding https://exhentai.org/g/1509162/87cf84cdae/ 

**Expected behavior**
To download from EH

**Logs**
[error.log](https://github.com/happypandax/happypandax/files/3790268/error.log)

**Environment:**
 Windows 10
Firefox 72
HPX 0.12.3

**Comment by twiddli** [comment]
Hello,

1. Do you have access to exhentai as per EHentai Login?
2. Are you aware of EHentai Downloader only supporting downloads using credits/gp atm.?
2. Do you have enough credits/gp?

**Comment by Cupbox** [comment]
1: Yes
2: Yes. I download using credits/gp normally
3: Yes.
![image](https://user-images.githubusercontent.com/13769570/67911138-552ce480-fb7d-11e9-9920-74ab5fc0718f.png)

**Comment by twiddli** [comment]
can you give me the url of the gallery in question?

**Comment by Cupbox** [comment]
It's a general problem present with all E-H galleries I've tried.
Taking https://exhentai.org/g/1509162/87cf84cdae/ for example gave me the log in the first post.

**Comment by twiddli** [comment]
Can you try downloading in debug mode for the `debug.log`?
You can also provide the `plugin.log` inside the EHentai Downloader plugin folder.

**Comment by Cupbox** [comment]
Here
[debug.log](https://github.com/happypandax/plugins/files/3799272/debug.log)
[EHentai Metadata plugin.log](https://github.com/happypandax/plugins/files/3799273/plugin.log)
[plugin.log](https://github.com/happypandax/plugins/files/3799276/plugin.log)

**Comment by twiddli** [comment]
Can you try download with the setting **Delete lefover files when download fails** set to false, then look into the folder `cache/download` for the file that was just download and check if it's valid?

**Comment by Cupbox** [comment]
Nope, seems to be invalid.
For https://exhentai.org/g/1510351/81a60c00a8/, I got this [NieR_ Yurimata.zip](https://github.com/happypandax/plugins/files/3800037/NieR_.Yurimata.zip)
[debug.log](https://github.com/happypandax/plugins/files/3800044/debug.2.log)
[plugin.log](https://github.com/happypandax/plugins/files/3800040/plugin.log)

**Comment by twiddli** [comment]
okay I think it might be a race condition, the file isn't getting prepared in time for the link to be available, maybe
i'll try and tweak things a little

**Comment by Cupbox** [comment]
Yes, I've noticed that. The download link takes around 10 seconds to be available for download but in HPX, the DL has already failed by then.
